### PR TITLE
Stop excluding files from package list...

### LINKED ISSUE #1079: Editables from pipfile are installed shallow
Dependencies of editables are not included in the lock file, so they do not get installed during a fresh `pipenv install`, this is due to the `no_deps` switch in https://github.com/kennethreitz/pipenv/blob/023696e46993bf55cd2807eae1801f8884a6357a/pipenv/cli.py#L775

A minimal example is demonstrated in the following gist
https://gist.github.com/PaulVisscher/f20d2d56fff66414b85ed77622a3aa66

##### Describe your environment

1. Debian GNU/Linux 8
1. Python version: Python 3.6.3
1. Pipenv version: 3.8.2

##### Expected result
Dependencies of editable packages are added to the lock file and installed

##### Actual result
Editable packages are installed shallow: dependencies of editable packages are not added to the lock file, and thus not installed.
```
pipenv shell
python
>>> import foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/dummy/foo/foo/__init__.py", line 1, in <module>
    import jsonschema
ModuleNotFoundError: No module named 'jsonschema'
```

##### Steps to replicate
 - Replicate the structure from the linked [gist](https://github.com/kennethreitz/pipenv/blob/023696e46993bf55cd2807eae1801f8884a6357a/pipenv/cli.py#L775) or download the included test folder from the [test-case.zip](https://github.com/kennethreitz/pipenv/files/1480495/test-case.zip).
- Run
```
pipenv install
pipenv shell
````
- Run
```
python
>>> import foo
````

### PR DESCRIPTION
**Pull request by techalchemy**
- Fixes #1079 and #909
- I don't know why I implemented this logic in the first place
- I might be an idiot

### CODE REVIEW

File: pipenv/project.py [hunk 1]
```diff
@@ -67,13 +67,11 @@ def _build_package_list(self, package_section):
                     else:
                         ps.update({k: v})
                 else:
-                    if not (is_installable_file(k) or is_installable_file(v) or
-                            any(file_prefix in v for file_prefix in ['path', 'file'])):
-                        ps.update({k: v})
+                    ps.update({k: v})
             else:
                 if not (any(is_vcs(i) for i in [k, v]) or
```
Comment (nateprewitt) [inline pipenv/project.py]: We really need a comment expressing what this is doing and why.

File: pipenv/project.py [hunk 2]
```diff
@@ -67,13 +67,11 @@ def _build_package_list(self, package_section):
...
-                        any(is_installable_file(i) for i in [k, v]) or
-                        any(is_valid_url(i) for i in [k, v])):
```
Comment (nateprewitt) [inline pipenv/project.py]: Any reasons we're deviating from the use of `i` in the other checks? They should probably all be `entry` or `i`.

Comment (techalchemy) [reply pipenv/project.py]: Yeah I was contemplating swapping everything to `entry` but I had to re-think all my logic anyway so I just left it as `i` across the board (which is not very descriptive but uh, sometimes it keeps our lines under 80 characters?)

**Comment by techalchemy** [comment]
CI is lying, transient failures.  Things are fine

**Comment by techalchemy** [comment]
Btw appveyor passes on my own account:

[Build status](https://ci.appveyor.com/project/techalchemy/pipenv/branch/bugfix/1079-shallow-editable-installs)

You can fix a lot of transient appveyor failures if you add the following to https://ci.appveyor.com/project/kennethreitz/pipenv/settings/build  => before build script => cmd:

```cmd
rmdir /s /q "c:\users\appveyor\appdata\local\temp\1\tmp*\configparser"
rmdir /s /q "c:\users\appveyor\appdata\local\pip-tools\cache"
```

**Pull request merged**